test/tests/functional/pbs_alps_inventory_check_hook.py

```diff
@@ -1,58 +1,213 @@
+                starttime=now,
+                max_attempts=10,
+                interval=2)
+            os.rename(
+                "/opt/cray/alps/default/bin/stat",
+                "/opt/cray/alps/default/bin/apstat")
+        else:
+            self.skipTest("This test can be run on simulator")
+
+    def test_xthostname(self):
+        """
+        Test when hook attempts to read the /etc/xthostname file to
+        determine Cray hostname, but the hostname file is missing.
+        """
+        now = time.time()
+        if self.platform == "craysim":
+            os.rename("/etc/xthostname", "/etc/xt")
+            self.mom.log_match(
+                "/etc/xthostname file found on this host",
+                starttime=now,
+                max_attempts=10,
+                interval=2)
+            os.rename("/etc/xt", "/etc/xthostname")
+        else:
+            self.skipTest("This test can be run on simulator")
+
+    def test_start_of_hook(self):
+        """
+        Test log at the start of hook processing.
+        """
+        now = time.time()
+        self.mom.log_match(
+            "Processing ALPS inventory for crayhost %s" % self.crayhostname,
+            starttime=now,
+            max_attempts=10,
+            interval=2)
+
+    def test_cray_login_nodes(self):
+        """
+        Test log when no nodes with vntype 'cray_login' are present.
+        """
+        if self.platform == "craysim":
+            now = time.time()
+            mc = self.mom.parse_config()
+            save = mc["$alps_client"]
+            del mc["$alps_client"]
+            self.mom.apply_config(mc)
+            self.mom = self.moms.values()[0]
+            self.host = self.mom.shortname
+            rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+            self.assertEqual(rc, 0)
+            rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.host)
+            self.assertEqual(rc, 0)
+            self.mom.log_match(
+                "ALPS Inventory Check: No eligible " +
+                "login nodes to perform inventory check",
+                starttime=now,
+                max_attempts=10,
+                interval=2)
+            mc["$alps_client"] = save
+            self.mom.apply_config(mc, False)
+
+    def delete_cray_compute_node(self):
+        """
+        Deletes the cray compute node from pbs node list
+        """
+        vnl = self.server.filter(
+            VNODE, {'resources_available.vntype': 'cray_compute'})
+        vlist = vnl["resources_available.vntype=cray_compute"]
+        vnode = vlist[0]
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, id=vnode)
+        self.assertEqual(rc, 0)
+
+    def test_pbs_home_path(self):
+        """
+        Test log when mom_priv directory is not in the expected/default
+        location (PBS_HOME), indicating a PBS installation issue.
+        """
+        if self.platform == "craysim":

```

Comment:
No in Cray etc directory is read only it can be modified through admin command only.